vnext/ReactUWP/Modules/DeviceInfoModule.cpp

```diff
@@ -59,11 +61,21 @@ void DeviceInfo::attachRoot(winrt::FrameworkElement rootElement) {
       updateRootElementSize(static_cast<float>(root.ActualWidth()), static_cast<float>(root.ActualHeight()));
     }
   });
+  auto displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
+  m_dpiChangedRevoker = displayInfo.DpiChanged(winrt::auto_revoke, [this, displayInfo](const auto &, const auto &) {
+    m_scale = static_cast<float>(displayInfo.ResolutionScale()) / 100;
+    m_dimensions["windowPhysicalPixels"]["scale"] = m_scale;
+    m_dimensions["windowPhysicalPixels"]["densityDpi"] = displayInfo.LogicalDpi();
+    m_dimensions["screenPhysicalPixels"]["scale"] = m_scale;
+    m_dimensions["screenPhysicalPixels"]["densityDpi"] = displayInfo.LogicalDpi();
+    updateRootElementSize(m_width, m_height);
+  });
```

Comment:
I suggest refactoring the code that updates m_dimensions into the update() method.  As it stands we now have 3 separate places that updates the same data.

Reply:
update() is actually initialize(), it creates the folly table for m_dimensions and fill in the data for every table entry.
When CoreWindow size changes, we only need to update the table for windowPhysicalPixels's width and height entries.
When DPI changes, we need to update scale and densityDpi entries for windowPhysicalPixels and screenPhysicalPixels and also needs to recalculate the windowPhysicalPixels's width and height entries.

We are missing a event to update fontScale entries, but it is out of scope for this bug.

I will rename update function and do some code cleanup..

---
In reply to: [373229116](https://github.com/microsoft/react-native-windows/pull/3956#discussion_r373229116) [](ancestors = 373229116)

Reply:
update() doesn't have to just be used for initializing -  it's totally fine to create a brand new folly object when things change.  In fact I already changed the code to use update when things change in a prototype branch. 

Please do clean this code up - it's difficult for a reader to follow all the code paths.

Reply:
Will clean it up and forget about optimization, this code path won't get hit frequently.

---
In reply to: [373237232](https://github.com/microsoft/react-native-windows/pull/3956#discussion_r373237232) [](ancestors = 373237232)